src/main/kotlin/no/nav/syfo/application/service/VarselService.kt

```diff
@@ -0,0 +1,22 @@
+package no.nav.syfo.application.service
+
+import no.nav.syfo.application.IVarselProducer
+import no.nav.syfo.application.IVarselRepository
+import no.nav.syfo.domain.Varsel
+
+class VarselService(
+    private val varselRepository: IVarselRepository,
+    private val varselProducer: IVarselProducer,
+) {
+    fun publishUnpublishedVarsler(): List<Result<Varsel>> {
+        val unpublishedVarsler = varselRepository.getUnpublishedVarsler()
+
+        return unpublishedVarsler.map { (personident, varsel) ->
+            runCatching {
+                varselProducer.sendArbeidstakerVarsel(personIdent = personident, varsel = varsel)
+                varselRepository.setPublished(varsel)
+                varsel
+            }
```

Comment:
Akkurat nå blir det usynlig for domenet at varsel blir publisert. Trenger sikkert ikke å gjøre nå, men tenker egentlig at det bør gjenspeiles i domenemodellen ettersom det er en viktig del av forretningslogikken🤔 Hva tenker du? Hadde kanskje blitt noe slikt? 👇 Mulig vi da heller ikke hadde trengt en spesifikk `setPublished` funksjon i repositoriet, men bare en `updateVarsel` funksjon.

```kotlin
        return unpublishedVarsler.map { (personident, varsel) ->
            runCatching {
                varselProducer.sendArbeidstakerVarsel(personIdent = personident, varsel = varsel)
		val publishedVarsel = varsel.publish()
                varselRepository.setPublished(publishedVarsel)
                varsel
            }
```

Reply:
Ja, tenkte på det. Kan godt gjøre det, men føler at en updateVarsel-funksjon da må begrense seg til de feltene vi faktisk kommer til å oppdatere, som bare er updated_at og published_at foreløpig (og journalpostId etter hvert).

Reply:
Hmm, ja. Vi kan godt vente med det😊 Også usikker på hvordan vi best mulig fletter sammen `producer`, `domain` og `repository` her slik at man "må" innom `varsel.publish()` for at man skal kunne oppdatere i databasen 🤔